app/controllers/ApprovalController.scala

```diff
@@ -46,38 +48,41 @@ class ApprovalController @Inject() (identify: IdentifierAction, approvalService:
       case Right(id) => Created(Json.obj("id" -> id))
       case Left(err @ Error(Error.UnprocessableEntity, _, Some(details))) =>
         logger.error(s"Failed to save process for approval due to process errors $details")
-        UnprocessableEntity(Json.toJson(err))
+        UnprocessableEntity(toJson(err))
+      case Left(DuplicateKeyError) =>
+        logger.error(s"Failed to save process for approval due to duplicate processCode")
+        UnprocessableEntity(toJson(Error(Error.UnprocessableEntity, "Duplicate ProcessCode - the process has the same processCode as an existing process")))
```

Comment:
I'm not sure BadRequest wouldn't be a better fit; the process could be fine, but we can't process it because a process with the same process code already exists.

Reply:
This was more to get the error message back to ocelot as it was using same structure for passing back errors as the duplicate page urls etc

Reply:
That did occur to me!